msteams-platform/sbs-create-dashboard-widget-graph-api-call-in-Teams-toolkit.yml

```diff
@@ -0,0 +1,283 @@
+
+      > [!NOTE]
+      > You can implement a service to retrieve data from the backend service or from Microsoft Graph API.
+    
+     ## Create a widget file
+
+     1. In Visual Studio Code, go to **EXPLORER** > **tabs** > **src** > **views** > **widgets**.
+     1. Create a **sampleWidget.ts** file.
+     1. Add the following code in the **sampleWidget.tsx** file:
+
+        ```typescript
+          import { Button, Text } from "@fluentui/react-components";
+          import { Widget } from "../lib/Widget";
+          import { SampleModel } from "../../models/sampleModel";
+          import { getSampleData } from "../../services/sampleService";
+
+          export class SampleWidget extends Widget<SampleModel> {
+              async getData(): Promise<SampleModel> {
+                  return getSampleData();
+              }
+
+              headerContent(): JSX.Element | undefined {
+                  return <Text>Sample Widget < /Text>;
+              }
+
+              bodyContent(): JSX.Element | undefined {
+                  return <div>{ this.state.data ? content} < /div>;
+              }
+
+              footerContent(): JSX.Element | undefined {
+                  return (<Button
+                      appearance= "primary"
+                  size = "medium"
+                  style = {{ width: "fit-content" }
+              }
+              onClick = {() => { }}
+                    >
+              View Details
+                  < /Button>);
+                    }
+                    }
+        ```
+
+        :::image type="content" source="~/assets/images/sbs-create-a-new-dashboard/create-sample-widget.png" alt-text="Screenshot shows the creation of sampleWidget.tsx under the widgets in Visual Studio Code.":::
+      
+     ## Add the widget to the dashboard
+
+      If you want to create a new dashboard, refer to [add a new dashboard](tabs/how-to/dashboard-teams-toolkit.md#add-a-new-dashboard).
+
+      1. In Visual Studio Code, go to **Explorer** > **tabs** > **src** > **views** > **dashboards** > **sampleDashboard.tsx**
+      1. Create a **sampleDashboard.jsx** file.
+      1. Add the following code in the **sampleDashboard.jsx** file:
+
+          ```jsx
+          protected dashboardLayout(): JSX. Element | undefined { 
+          return (
+          <>
+          <ListWidget />
+          <ChartWidget />
+          <SampleWidget />
+          </>);
+          } 
+          ```
+
+          ---
+
+          :::image type="content" source="~/assets/images/sbs-create-a-new-dashboard/create-sample-dashboard.png" alt-text="Screenshot shows the existing sampleDashboard file in Visual Studio Code."::: 
+
+          If you want the widget in a column, you can use the `oneColumn()` method to define the column layout.
+      
+          Following is the sample of `oneColumn()` layout:
+
+              ```jsx
```

Comment:
There is a problem with this code snippet layout.

Reply:
Updated.